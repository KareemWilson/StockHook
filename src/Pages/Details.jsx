import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router';
import { getDetails } from '../redux/details/details';

function Details() {
  const { symbol, name } = useParams();
  const details = useSelector((state) => state.details);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getDetails(symbol));
  }, [dispatch]);

  return (
    <div className="details-container d-flex flex-column align-items-center ">
      <p className="name-of-company-in-details">{name}</p>
      <div className="details">
        {Object.keys(details).map((field) => (
          <div
            key={field}
            className="detail-field d-flex flex-row justify-content-between px-3 align-items-center"
          >
            <p className="field-name">{field}</p>
            <p className="field-data">{details[field]}</p>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Details;
