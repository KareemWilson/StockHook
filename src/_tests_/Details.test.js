import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Details from '../Pages/Details';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';

describe('home page rendering', () => {
  test('Render home page correctly', () => {
    const homePage = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Details />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(homePage).toMatchSnapshot();
  });
});
