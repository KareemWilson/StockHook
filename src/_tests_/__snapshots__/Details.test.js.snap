// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page rendering Render home page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details-container d-flex flex-column align-items-center "
      >
        <a
          href="/"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-left back-icon"
            data-icon="arrow-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
        </a>
        <p
          class="name-of-company-in-details"
        />
        <div
          class="details"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details-container d-flex flex-column align-items-center "
    >
      <a
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-left back-icon"
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
      </a>
      <p
        class="name-of-company-in-details"
      />
      <div
        class="details"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
